src/cli/generator/generator.rs

```diff
@@ -95,18 +95,27 @@ impl Generator {
     async fn resolve_io(&self, config: Config<Resolved>) -> anyhow::Result<Vec<Input>> {
         let mut input_samples = vec![];
 
-        let reader = ResourceReader::cached(self.runtime.clone());
-        let proto_reader = ProtoReader::init(reader.clone(), self.runtime.clone());
+        let cached_reader = ResourceReader::cached(self.runtime.clone());
+        let direct = Direct::init(self.runtime.clone());
```

Comment:
1. Stick to the same resource reader. 
2. Implement `Hash` on `Resource` for building stable hash keys.

Reply:
done